### ISSUE TITLE
Expensive PipelineArgumentsTab.updatePipelineOptionsForm() is called twice when changing radio selection

### ISSUE THREAD

**Issue by chanseokoh** [GoogleCloudPlatform/google-cloud-eclipse#2651]
We add the selection-changed listener below to the radio buttons. As the Javadoc says, the intention is to update the tab when the runner selection is changed.

```java
  /**
   * When the Runner selection is changed, update the underlying launch configuration, update the
   * PipelineOptionsForm to show all available inputs, and re-render the tab.
   */
  private class UpdateLaunchConfigAndRequiredArgsSelectionListener extends SelectionAdapter {
    ...
    public void widgetSelected(SelectionEvent e) {
      launchConfiguration.setRunner(runner);
      updatePipelineOptionsForm();
    }
  }
```

However, `widgetSelected()` is called not only when a button is selected but also when a button is de-selected. Since `updatePipelineOptionsForm()` is very expensive, we should basically do

```java
    @Override
    public void widgetSelected(SelectionEvent e) {
      Button radio = (Button) e.getSource();
      if (radio.getSelection()) {
        launchConfiguration.setRunner(runner);
        updatePipelineOptionsForm();
      }
    }
```

**Comment by briandealwis** [comment]
ZOMG good find: this is why we see these _Pipeline Options Type XXX not found_ warnings logged (#2232).

**Comment by chanseokoh** [comment]
Fixed by #2652.
